class.jetpack.php

```diff
@@ -3930,10 +3930,11 @@ function admin_page_load() {
 			Jetpack::restate();
 		}
 
+		$from = ! empty( $_GET['from'] ) ? $_GET['from'] : 'iframe';
```

Comment:
Should we do any validation that the value is an URL? Made a cursory look and see we often use `build_connect_url` with the `$raw = false` parameter so not escaped. I think they're all passed to `wp_redirect` or escaped later, but didn't exhaustively look.

Otherwise, I'm good with it once we confirm we're happy with not validating the URL.

Reply:
FWIW the `from` value isn't really an URL, I've done a quick search and it seems that none of the top 100 occurrences are URLs. Here's top 10 for example:

* full-screen-prompt
* unknown
* landing-page-top
* woocommerce-setup-wizard
* iframe
* widget-btn
* banner
* banner-44-slide-1
* banner-44-slide-1-plugins
* banner-44-slide-1-dashboard

Reply:
My bad, it seemed like they were all used in the URL and my mind jumped to that. Thanks.